DAOS-7392 daos: add preserve_props option to fs copy

dsikich opened the pull request:
* add an option to preserve container properties
	  to filesystem copy tool
	* add check for container status before copying
	* add serialization support for DAOS_PROP_CO_ROOTS
	  and DAOS_PROP_CO_EC_CELL_SZ
	* add functional tests for the preserve props option

The preserve properties option within filesystem copy can be
used when moving data from DAOS->POSIX and then POSIX->DAOS.
If moving data from DAOS->POSIX the container properties would
be written to an hdf5 file, and then when moving data back
to DAOS the path to the file can be provided so that the newly
created DAOS container will read and use the preserved DAOS
container properties within the hdf5 file.

Signed-off-by: Danielle Sikich <danielle.sikich@intel.com>

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

FYI: Errors found in lines not modified in the patch:

[src/tests/ftest/util/data_mover_test_base.py:1](https://github.com/daos-stack/daos/blob/64c58e786963d99228bf56a32753decffc93f874/src/tests/ftest/util/data_mover_test_base.py#L1):
(pylint-too-many-lines) Too many lines in module (1137/1000)

[src/tests/ftest/util/data_mover_utils.py:75](https://github.com/daos-stack/daos/blob/64c58e786963d99228bf56a32753decffc93f874/src/tests/ftest/util/data_mover_utils.py#L75):
(pylint-consider-using-dict-items) Consider iterating with .items()

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

FYI: Errors found in lines not modified in the patch:

[src/tests/ftest/util/data_mover_test_base.py:1](https://github.com/daos-stack/daos/blob/c357bbb505f43fcbb18114e8cf407dae236ba59d/src/tests/ftest/util/data_mover_test_base.py#L1):
(pylint-too-many-lines) Too many lines in module (1137/1000)

[src/tests/ftest/util/data_mover_utils.py:75](https://github.com/daos-stack/daos/blob/c357bbb505f43fcbb18114e8cf407dae236ba59d/src/tests/ftest/util/data_mover_utils.py#L75):
(pylint-consider-using-dict-items) Consider iterating with .items()

daosbuild1 commented:
Test stage Unit Test completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-6617/2/execution/node/790/log

daosbuild1 commented:
Test stage Unit Test completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-6617/3/execution/node/277/log

daosbuild1 commented:
Test stage Unit Test completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-6617/4/execution/node/255/log

daltonbohning commented:
> [src/tests/ftest/util/data_mover_test_base.py:1](https://github.com/daos-stack/daos/blob/c357bbb505f43fcbb18114e8cf407dae236ba59d/src/tests/ftest/util/data_mover_test_base.py#L1):
> (pylint-too-many-lines) Too many lines in module (1137/1000)

@dsikich We can ignore this one for now. It's not harmful, and really it requires abstracting more of this code elsewhere. In general, there are some functions in this file that could be moved/integrated at lower levels of the framework, which is what my plan was when I was working on datamover tests. If you ever feel like refactoring, let me know ;)

dsikich commented on src/utils/daos_hdlr.h:
```diff
@@ -190,6 +191,14 @@ int pool_autotest_hdlr(struct cmd_args_s *ap);
  */
 
+/* general datamover operations */
+void dm_cont_free_usr_attrs(int n, char ***_names, void ***_buffers, size_t **_sizes);
+int dm_cont_get_usr_attrs(struct cmd_args_s *ap, daos_handle_t coh, int *_n, char ***_names,
+			  void ***_buffers, size_t **_sizes);
+int dm_cont_get_all_props(struct cmd_args_s *ap, daos_handle_t coh, daos_prop_t **_props,
+			  bool get_oid, bool get_label, bool get_roots);
+int dm_copy_usr_attrs(struct cmd_args_s *ap, daos_handle_t src_coh, daos_handle_t dst_coh);
+
```
@daltonbohning these are wrappers that look up the symbol in the library, you cannot call them directly.

daltonbohning replied:
@dsikich But they are being defined in `daos-hdlr.c` as well?
https://github.com/daos-stack/daos/blob/c357bbb505f43fcbb18114e8cf407dae236ba59d/src/utils/daos_hdlr.c#L2502-L2517

dsikich replied:
@daltonbohning these are not actually defined in daos_serialize.c because they do not require HDF5 so I was trying to limit how many functions I had to lookup in the library with dlsym. I think they could either be "wrappers" where they are defined in daos_serialize.c then I would have to have to look them up, or I just defined them directly in daos_hdlr.c.

dsikich replied:
@daltonbohning I also remember now I wanted to use some of these functions in the case that serialization was not being used. For instance to copy properties when data is being transferred from DAOS->DAOS.

daltonbohning replied:
Oh, I see... For some reason I thought they had been defined in daos_serialize.c. Please disregard 👍

dsikich commented on src/client/serialize/daos_serialize.c:
```diff
@@ -8,13 +8,88 @@
+ * hi, lo uint64_t's. In order to write these to an hdf5 file a similar
+ * type has to be defined to describe/store it to hdf5
+ */
+struct obj_id {
+    uint64_t hi;
+    uint64_t lo;
+};
```
@daltonbohning I tried and it didn't work when I did try to use it directly. This is a definition for inside of the hdf5 file, which doesn't recognize the daos_obj_id_t type in the file. This is why the acl is actually written as a list of strings because you cannot write to the file as a daos_acl type, hdf5 does not know how that is represented in DAOS.

daltonbohning replied:
@dsikich `daos_prop_t` makes sense, because `daos_prop_t.dpp_entries` is an array. But here, `struct obj_id` is almost the same as `daos_obj_id_t`, and `daos_obj_id_t` is public, so I would think that should work. But I'm not familiar enough with HDF5

dsikich replied:
@daltonbohning oh oops I see what you mean use the type that was defined to write the OIDS, yeah I can probably do that. I misunderstood and that you meant the native DAOS type.

dsikich commented on src/utils/SConscript:
```diff
@@ -8,7 +8,7 @@ def scons():
         return
 
-    libs = ['daos', 'daos_common', 'uuid', 'dfs', 'duns', 'gurt', 'cart']
+    libs = ['daos', 'daos_common', 'uuid', 'dfs', 'duns', 'gurt', 'cart', 'dl']
```
@daltonbohning yeah I asked about this a while back but can add more people. The dl library is added in other SConscript files in DAOS so it is already a dependency elsewhere.

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

FYI: Errors found in lines not modified in the patch:

[src/tests/ftest/util/data_mover_test_base.py:1](https://github.com/daos-stack/daos/blob/2f38387ddf1dfc1119f3d87773192173a7676163/src/tests/ftest/util/data_mover_test_base.py#L1):
(pylint-too-many-lines) Too many lines in module (1137/1000)

[src/tests/ftest/util/data_mover_utils.py:75](https://github.com/daos-stack/daos/blob/2f38387ddf1dfc1119f3d87773192173a7676163/src/tests/ftest/util/data_mover_utils.py#L75):
(pylint-consider-using-dict-items) Consider iterating with .items()

daosbuild1 commented:
Test stage Unit Test completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-6617/5/execution/node/751/log

daosbuild1 commented:
Test stage NLT completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-6617/5/execution/node/847/log

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

FYI: Errors found in lines not modified in the patch:

[src/tests/ftest/util/data_mover_test_base.py:1](https://github.com/daos-stack/daos/blob/31c35ffd7e2b59bd6aa5a5fb1899200112515a24/src/tests/ftest/util/data_mover_test_base.py#L1):
(pylint-too-many-lines) Too many lines in module (1137/1000)

[src/tests/ftest/util/data_mover_utils.py:75](https://github.com/daos-stack/daos/blob/31c35ffd7e2b59bd6aa5a5fb1899200112515a24/src/tests/ftest/util/data_mover_utils.py#L75):
(pylint-consider-using-dict-items) Consider iterating with .items()

daosbuild1 commented:
Test stage Unit Test completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-6617/6/execution/node/737/log

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

FYI: Errors found in lines not modified in the patch:

[src/tests/ftest/util/data_mover_test_base.py:1](https://github.com/daos-stack/daos/blob/694dd2ba9fa17a7b038ba5e5a8a2a41ea0ef7347/src/tests/ftest/util/data_mover_test_base.py#L1):
(pylint-too-many-lines) Too many lines in module (1137/1000)

[src/tests/ftest/util/data_mover_utils.py:75](https://github.com/daos-stack/daos/blob/694dd2ba9fa17a7b038ba5e5a8a2a41ea0ef7347/src/tests/ftest/util/data_mover_utils.py#L75):
(pylint-consider-using-dict-items) Consider iterating with .items()

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

FYI: Errors found in lines not modified in the patch:

[src/tests/ftest/util/data_mover_test_base.py:1](https://github.com/daos-stack/daos/blob/14a228a3b35231652cb65d54e3b311eac31fa81c/src/tests/ftest/util/data_mover_test_base.py#L1):
(pylint-too-many-lines) Too many lines in module (1137/1000)

[src/tests/ftest/util/data_mover_utils.py:75](https://github.com/daos-stack/daos/blob/14a228a3b35231652cb65d54e3b311eac31fa81c/src/tests/ftest/util/data_mover_utils.py#L75):
(pylint-consider-using-dict-items) Consider iterating with .items()

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

FYI: Errors found in lines not modified in the patch:

[src/tests/ftest/util/data_mover_test_base.py:1](https://github.com/daos-stack/daos/blob/4d6bb9c290685bbb78bf90da4fe529b8c9fca61f/src/tests/ftest/util/data_mover_test_base.py#L1):
(pylint-too-many-lines) Too many lines in module (1137/1000)

[src/tests/ftest/util/data_mover_utils.py:75](https://github.com/daos-stack/daos/blob/4d6bb9c290685bbb78bf90da4fe529b8c9fca61f/src/tests/ftest/util/data_mover_utils.py#L75):
(pylint-consider-using-dict-items) Consider iterating with .items()

dsikich commented:
I created a ticket here: https://daosio.atlassian.net/browse/DAOS-8528

for the NLT warning because it revealed a bug when passing the ACL property to container create it is leaking memory in daos_acl_dup on the server

daltonbohning commented:
> [src/tests/ftest/util/data_mover_utils.py:75](https://github.com/daos-stack/daos/blob/4d6bb9c290685bbb78bf90da4fe529b8c9fca61f/src/tests/ftest/util/data_mover_utils.py#L75):
> (pylint-consider-using-dict-items) Consider iterating with .items()

I do think we should follow this one so we don't have noise.

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

FYI: Errors found in lines not modified in the patch:

[src/tests/ftest/util/data_mover_test_base.py:1](https://github.com/daos-stack/daos/blob/742917e9b478ab8b39241a6de2ae5c0790eda938/src/tests/ftest/util/data_mover_test_base.py#L1):
(pylint-too-many-lines) Too many lines in module (1137/1000)

daosbuild1 commented:
Test stage Build RPM on CentOS 8 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-6617/10/execution/node/399/log

daosbuild1 commented:
Test stage Build RPM on Leap 15 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-6617/10/execution/node/378/log

daosbuild1 commented:
Test stage Build RPM on CentOS 7 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-6617/10/execution/node/375/log

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

FYI: Errors found in lines not modified in the patch:

[src/tests/ftest/util/data_mover_test_base.py:1](https://github.com/daos-stack/daos/blob/742917e9b478ab8b39241a6de2ae5c0790eda938/src/tests/ftest/util/data_mover_test_base.py#L1):
(pylint-too-many-lines) Too many lines in module (1137/1000)

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

FYI: Errors found in lines not modified in the patch:

[src/tests/ftest/util/data_mover_test_base.py:1](https://github.com/daos-stack/daos/blob/d151dc3b7c0a0d71d9ce995556b916f259b5e4a8/src/tests/ftest/util/data_mover_test_base.py#L1):
(pylint-too-many-lines) Too many lines in module (1137/1000)

daosbuild1 commented:
Test stage Functional Hardware Medium completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-6617/12/display/redirect

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

FYI: Errors found in lines not modified in the patch:

[ci/functional/required_packages.sh:6](https://github.com/daos-stack/daos/blob/15a5e69d5f6f8cd0ed6a3910c14c04ae9ec42370/ci/functional/required_packages.sh#L6):
(lint) client_ver appears unused. Verify use (or export if used externally). [SC2034]

[src/tests/ftest/util/data_mover_test_base.py:1](https://github.com/daos-stack/daos/blob/15a5e69d5f6f8cd0ed6a3910c14c04ae9ec42370/src/tests/ftest/util/data_mover_test_base.py#L1):
(pylint-too-many-lines) Too many lines in module (1137/1000)

daosbuild1 commented:
Test stage Functional Hardware Large completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-6617/13/display/redirect

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

FYI: Errors found in lines not modified in the patch:

[ci/functional/required_packages.sh:6](https://github.com/daos-stack/daos/blob/1855043e95379021e37d5c4ad35baa62508c74e8/ci/functional/required_packages.sh#L6):
(lint) client_ver appears unused. Verify use (or export if used externally). [SC2034]

[src/tests/ftest/util/data_mover_test_base.py:1](https://github.com/daos-stack/daos/blob/1855043e95379021e37d5c4ad35baa62508c74e8/src/tests/ftest/util/data_mover_test_base.py#L1):
(pylint-too-many-lines) Too many lines in module (1137/1000)

daltonbohning commented on src/tests/ftest/datamover/dm_posix_preserve_props.py:
```diff
@@ -178,7 +178,7 @@ def verify_cont_prop(self, cont, prop_list, api):
         for prop_idx, prop in enumerate(prop_list):
             # This one is not set
-            if api == "DFS" and "OID" in prop_list[prop_idx]:
+            if api == "DFS" and ("OID" or "ROOTS") in prop_list[prop_idx]:
```
I think `("OID" or "ROOTS") in prop_list[prop_idx]` will always evaluate to:
`"OID" in prop_list[prop_idx]`

```
>>> my_s = "aaa ROOTS bbb"
>>> ("OID" or "ROOTS")
'OID'
>>> ("OID" or "ROOTS") in my_s
False
>>> "ROOTS" in my_s
True
```

daltonbohning reviewed (commented):
FYI, there is a conflict in `src/utils/daos.c`

daosbuild1 commented on utils/rpms/daos.spec:
```diff
@@ -474,10 +474,13 @@ getent passwd daos_agent >/dev/null || useradd -s /sbin/nologin -r -g daos_agent
 
 %changelog
+* Tue Sep 21 2021 Danielle M. Sikich <danielle.sikich@intel.com> 1.3.105-3
+- Fix name of daos serialize package 
```
(style)  trailing whitespace

daosbuild1 reviewed (changes requested):
Style warning(s) for job https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-6617/16/
Please review https://wiki.hpdd.intel.com/display/DC/Coding+Rules

FYI: Errors found in lines not modified in the patch:

[ci/functional/required_packages.sh:6](https://github.com/daos-stack/daos/blob/240c94c115fa2a9cfdee0e463c2d18995cb285e1/ci/functional/required_packages.sh#L6):
(lint) client_ver appears unused. Verify use (or export if used externally). [SC2034]

[src/tests/ftest/util/data_mover_test_base.py:1](https://github.com/daos-stack/daos/blob/240c94c115fa2a9cfdee0e463c2d18995cb285e1/src/tests/ftest/util/data_mover_test_base.py#L1):
(pylint-too-many-lines) Too many lines in module (1137/1000)

daosbuild1 commented:
Test stage checkpatch completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-6617/16/execution/node/120/log

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

FYI: Errors found in lines not modified in the patch:

[ci/functional/required_packages.sh:6](https://github.com/daos-stack/daos/blob/c3ab5bb0a85bb1fb61ea39e4a6abdde9235522c1/ci/functional/required_packages.sh#L6):
(lint) client_ver appears unused. Verify use (or export if used externally). [SC2034]

[src/tests/ftest/util/data_mover_test_base.py:1](https://github.com/daos-stack/daos/blob/c3ab5bb0a85bb1fb61ea39e4a6abdde9235522c1/src/tests/ftest/util/data_mover_test_base.py#L1):
(pylint-too-many-lines) Too many lines in module (1137/1000)

daosbuild1 commented:
Test stage Unit Test completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-6617/17/execution/node/697/log

daosbuild1 commented:
Test stage Unit Test completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-6617/18/execution/node/255/log

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

FYI: Errors found in lines not modified in the patch:

[ci/functional/required_packages.sh:6](https://github.com/daos-stack/daos/blob/3f4401026e07518b40979a37a183741e9eb129d7/ci/functional/required_packages.sh#L6):
(lint) client_ver appears unused. Verify use (or export if used externally). [SC2034]

[src/tests/ftest/util/data_mover_test_base.py:1](https://github.com/daos-stack/daos/blob/3f4401026e07518b40979a37a183741e9eb129d7/src/tests/ftest/util/data_mover_test_base.py#L1):
(pylint-too-many-lines) Too many lines in module (1137/1000)

daosbuild1 commented:
Test stage Functional Hardware Medium completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-6617/29/execution/node/1532/log

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

FYI: Errors found in lines not modified in the patch:

[ci/functional/required_packages.sh:6](https://github.com/daos-stack/daos/blob/35efd8d142d3152dd0e25466a7fad45d7c57d392/ci/functional/required_packages.sh#L6):
(lint) client_ver appears unused. Verify use (or export if used externally). [SC2034]

[src/tests/ftest/util/data_mover_test_base.py:1](https://github.com/daos-stack/daos/blob/35efd8d142d3152dd0e25466a7fad45d7c57d392/src/tests/ftest/util/data_mover_test_base.py#L1):
(pylint-too-many-lines) Too many lines in module (1199/1000)

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

FYI: Errors found in lines not modified in the patch:

[ci/functional/required_packages.sh:6](https://github.com/daos-stack/daos/blob/224672ce96d7fb54ca5d8005f790c50818b505c7/ci/functional/required_packages.sh#L6):
(lint) client_ver appears unused. Verify use (or export if used externally). [SC2034]

[src/tests/ftest/util/data_mover_test_base.py:1](https://github.com/daos-stack/daos/blob/224672ce96d7fb54ca5d8005f790c50818b505c7/src/tests/ftest/util/data_mover_test_base.py#L1):
(pylint-too-many-lines) Too many lines in module (1199/1000)

dsikich commented on utils/rpms/daos.spec:
```diff
@@ -509,17 +509,15 @@ getent passwd daos_agent >/dev/null || useradd -s /sbin/nologin -r -g daos_agent
-- Rename daos-tests daos-client-tests and make daos-tests require all
-  other test suites to maintain existing behavior
+* Mon Oct 28 2021 Danielle M. Sikich <danielle.sikich@intel.com> 2.1.100-5
+- Fix name of daos serialize package
```
@daltonbohning thanks I think I had a bad merge somewhere so I updated this

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

FYI: Errors found in lines not modified in the patch:

[ci/functional/required_packages.sh:6](https://github.com/daos-stack/daos/blob/23c1701aedecf3e0d49afefbfb1de6a4ed72aeda/ci/functional/required_packages.sh#L6):
(lint) client_ver appears unused. Verify use (or export if used externally). [SC2034]

[src/tests/ftest/util/data_mover_test_base.py:1](https://github.com/daos-stack/daos/blob/23c1701aedecf3e0d49afefbfb1de6a4ed72aeda/src/tests/ftest/util/data_mover_test_base.py#L1):
(pylint-too-many-lines) Too many lines in module (1199/1000)

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

FYI: Errors found in lines not modified in the patch:

[ci/functional/required_packages.sh:6](https://github.com/daos-stack/daos/blob/cad1df94dbf0073c46928e0e20424cf962e939e2/ci/functional/required_packages.sh#L6):
(lint) client_ver appears unused. Verify use (or export if used externally). [SC2034]

[src/tests/ftest/util/data_mover_test_base.py:1](https://github.com/daos-stack/daos/blob/cad1df94dbf0073c46928e0e20424cf962e939e2/src/tests/ftest/util/data_mover_test_base.py#L1):
(pylint-too-many-lines) Too many lines in module (1199/1000)

mchaarawi commented on src/tests/ftest/datamover/posix_preserve_props.py:
```diff
@@ -0,0 +1,240 @@
+            when API is unknown, including
+            container properties and user attributes.
+
+        :avocado: tags=all,full_regression
```
can't we add this to daily regression at least?

dsikich replied:
@mchaarawi https://daosio.atlassian.net/browse/DAOS-9001 I created a ticket I can do this as soon as this merges which should be fairly quick if I just update the test file.

dsikich replied:
@mchaarawi closed that ticket and just added it to daily_regression since I had to update PR anyway.

ashleypittman commented on src/client/serialize/daos_serialize.c:
```diff
@@ -524,9 +592,79 @@ deserialize_uint(hid_t file_id, uint64_t *val, const char *prop_str)
+		D_ERROR("failed to get dimensions of dataspace\n");
+		D_GOTO(out, rc = -DER_MISC);
+	}
+	D_ALLOC(roots, sizeof(struct daos_prop_co_roots));
```
```suggestion
	D_ALLOC_PTR(roots);
```

ashleypittman commented on src/utils/daos_hdlr.c:
```diff
@@ -1869,46 +1872,417 @@ set_dm_args_default(struct dm_args *dm)
+	D_ALLOC(names, num_attrs * sizeof(char *));
+	if (names == NULL)
+		D_GOTO(out, rc = -DER_NOMEM);
+	D_ALLOC(sizes, num_attrs * sizeof(size_t));
```
D_ALLOC_ARRAY?

ashleypittman commented on src/utils/daos_hdlr.c:
```diff
@@ -1869,46 +1872,417 @@ set_dm_args_default(struct dm_args *dm)
+	*_buffers = buffers;
+	*_sizes = sizes;
+out:
+	if (rc != 0) {
+		dm_cont_free_usr_attrs(num_attrs, &names, &buffers, &sizes);
+	}
```
```suggestion
	if (rc != 0)
		dm_cont_free_usr_attrs(num_attrs, &names, &buffers, &sizes);
```

ashleypittman commented on src/utils/daos_hdlr.c:
```diff
@@ -1869,46 +1872,417 @@ set_dm_args_default(struct dm_args *dm)
+	*_props = props;
+out:
+	daos_prop_free(prop_acl);
+	if (rc != 0) {
+		daos_prop_free(props);
 	}
+	return rc;
```
```suggestion
	if (rc != 0)
		daos_prop_free(props);
	return rc;
```

ashleypittman commented on src/utils/daos_hdlr.c:
```diff
@@ -1869,46 +1872,417 @@ set_dm_args_default(struct dm_args *dm)
+static int
+dm_check_cont_status(struct cmd_args_s *ap, daos_handle_t coh, bool *status_healthy)
+{
+	daos_prop_t		*prop = NULL;
```
```suggestion
	daos_prop_t		*prop;
```

ashleypittman commented on src/utils/daos_hdlr.c:
```diff
@@ -1953,6 +2326,16 @@ dm_connect(struct cmd_args_s *ap,
+			DH_PERROR_DER(ap, rc, "Failed to check container status");
+			D_GOTO(err, rc);
+		} else if (!status_healthy) {
+			DH_PERROR_DER(ap, rc, "Container status is unhealthy, stopping");
+			D_GOTO(err, rc);
+		}
```
rc will be 0 here, so DH_PERROR_DER is wrong, perhaps you need to assign it.
```suggestion
                         rc = -DER_INVAL;
			DH_PERROR_DER(ap, rc, "Container status is unhealthy, stopping");
			D_GOTO(err, rc);
		}
```

ashleypittman reviewed (commented):
We have a fault injection test for fs copy I think, we could easily add props to that and get some comprehensive coverage of this code.

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

FYI: Errors found in lines not modified in the patch:

[ci/functional/required_packages.sh:6](https://github.com/daos-stack/daos/blob/762a85423098b86ded8dd9cf1c6e2ff554f44e65/ci/functional/required_packages.sh#L6):
(lint) client_ver appears unused. Verify use (or export if used externally). [SC2034]

[src/tests/ftest/util/data_mover_test_base.py:1](https://github.com/daos-stack/daos/blob/762a85423098b86ded8dd9cf1c6e2ff554f44e65/src/tests/ftest/util/data_mover_test_base.py#L1):
(pylint-too-many-lines) Too many lines in module (1199/1000)

dsikich commented:
> We have a fault injection test for fs copy I think, we could easily add props to that and get some comprehensive coverage of this code.

@ashleypittman I addressed your other comments and added a ticket for this specifically: https://daosio.atlassian.net/browse/DAOS-9010

I am not sure if it matters for fault injection testing that this particular fs copy option requires the daos_serialize.so library (which requires HDF5).

ashleypittman commented on src/utils/daos_hdlr.c:
```diff
@@ -2451,46 +2453,416 @@ set_dm_args_default(struct dm_args *dm)
+	/* Sanity check */
+	if (num_attrs == 0) {
+		DH_PERROR_DER(ap, rc, "Failed to parse user attributes");
+		D_GOTO(out, rc = -DER_MISC);
```
```suggestion
		D_GOTO(out, rc = -DER_INVAL);
```

ashleypittman commented on src/utils/daos_hdlr.c:
```diff
@@ -1869,46 +1872,415 @@ set_dm_args_default(struct dm_args *dm)
+		/* ACL will be appended to the end */
+		props_merged = daos_prop_merge(props, prop_acl);
+		if (props_merged == NULL) {
+			DH_PERROR_DER(ap, rc, "Failed set container ACL");
+			D_GOTO(out, rc = -DER_INVAL);
```
```suggestion
			rc = -DER_INVAL;
			DH_PERROR_DER(ap, rc, "Failed set container ACL");
			D_GOTO(out, rc);
```

ashleypittman commented on src/utils/daos_hdlr.c:
```diff
@@ -1869,46 +1872,415 @@ set_dm_args_default(struct dm_args *dm)
+	/* Get all user attributes if any exist */
+	rc = dm_cont_get_usr_attrs(ap, ca->src_coh, &num_attrs, &names, &buffers, &sizes);
+	if (rc != 0) {
+		rc = 1;
```
This can't be right?  You're passing this value to a function that expect a daos error code later, would it be enough to simply remove this line?

dsikich replied:
@ashleypittman thanks, yeah I missed this one.

ashleypittman commented on src/tests/ftest/util/daos_utils.py:
```diff
@@ -632,14 +632,15 @@ def object_query(self, pool, cont, oid, sys_name=None):
         return data
 
-    def filesystem_copy(self, src, dst):
+    def filesystem_copy(self, src, dst, preserve_props=None):
```
This comment applies to several places in the python but shouldn't this be a True/False, what's the reason for using None here?

dsikich replied:
@ashleypittman honestly I don't have a strong reason. I was recently learning how to navigate and add tests here so I followed what it looks like most of the other tests do, which is set the parameter to None when it is a non-required argument.

daltonbohning replied:
`preserve_props` is actually a path it looks like. So I interpret `None` to mean "do not preserve" and `! None` to mean "do preserve, with this path"

dsikich replied:
yeah it is a an option that when used takes a path. I suppose there could have been a true/false preserve_props option but it would still require passing in the path if the option was used.

ashleypittman replied:
Ah, that makes sense.  None is the correct default then.

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

FYI: Errors found in lines not modified in the patch:

[ci/functional/required_packages.sh:6](https://github.com/daos-stack/daos/blob/c5d67378cfdc479ec825451f4ad894f9519bf5b6/ci/functional/required_packages.sh#L6):
(lint) client_ver appears unused. Verify use (or export if used externally). [SC2034]

[src/tests/ftest/util/data_mover_test_base.py:1](https://github.com/daos-stack/daos/blob/c5d67378cfdc479ec825451f4ad894f9519bf5b6/src/tests/ftest/util/data_mover_test_base.py#L1):
(pylint-too-many-lines) Too many lines in module (1199/1000)

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

FYI: Errors found in lines not modified in the patch:

[ci/functional/required_packages.sh:6](https://github.com/daos-stack/daos/blob/4dc43334a5ab2556093bf2d4e57a74629a6ee859/ci/functional/required_packages.sh#L6):
(lint) client_ver appears unused. Verify use (or export if used externally). [SC2034]

[src/tests/ftest/util/data_mover_test_base.py:1](https://github.com/daos-stack/daos/blob/4dc43334a5ab2556093bf2d4e57a74629a6ee859/src/tests/ftest/util/data_mover_test_base.py#L1):
(pylint-too-many-lines) Too many lines in module (1199/1000)

kprantis commented:
Not clear to me what the requiredness/need for 2.0 is for this one since it seems to have a ticket number for a resolved ticket?

dsikich commented:
> Not clear to me what the requiredness/need for 2.0 is for this one since it seems to have a ticket number for a resolved ticket?

@kprantis Thanks, looks like that was using the wrong ticket number. It should have been DAOS-7392.

ashleypittman commented on src/tests/ftest/datamover/posix_preserve_props.py:
```diff
@@ -0,0 +1,240 @@
+        pool1.connect(2)
+
+        # set the path to read and write container properties
+        self.preserve_props_path = self.tmp + "/cont_props.h5"
```
```suggestion
        self.preserve_props_path = os.path.join(self.tmp, "cont_props.h5")
```

ashleypittman commented on src/tests/ftest/datamover/posix_preserve_props.py:
```diff
@@ -0,0 +1,240 @@
+        src_props = self.write_cont(cont1)
+
+        # Create a posix source path
+        posix_path = join(self.new_posix_test_path(), self.test_file)
```
This might want to be os.path.join as well, I can't easily tell.

dsikich replied:
@ashleypittman this is using os.path.join from the import path at top of python file:
from os.path import join

ashleypittman commented on src/utils/daos_hdlr.c:
```diff
@@ -1869,46 +1872,416 @@ set_dm_args_default(struct dm_args *dm)
+
+	/* Sanity check */
+	if (num_attrs == 0) {
+		DH_PERROR_DER(ap, rc, "Failed to parse user attributes");
+		D_GOTO(out, rc = -DER_INVAL);
```
```suggestion
		rc = -DER_INVAL;
		DH_PERROR_DER(ap, rc, "Failed to parse user attributes");
		D_GOTO(out, rc);
```

ashleypittman commented on src/utils/daos_hdlr.c:
```diff
@@ -1869,46 +1872,416 @@ set_dm_args_default(struct dm_args *dm)
+		}
+		daos_prop_free(props);
+		props = props_merged;
+	} else if (rc && rc != -DER_NO_PERM) {
```
Just trying to understand the logic here, wouldn't this be the same and easier to read?
```suggestion
	} else if (rc != -DER_NO_PERM) {
```

daltonbohning commented on src/tests/ftest/datamover/posix_preserve_props.py:
```diff
@@ -58,7 +58,7 @@ def run_dm_preserve_props(self, tool, cont_type, api):
 
         # set the path to read and write container properties
-        self.preserve_props_path = self.tmp + "/cont_props.h5"
+        self.preserve_props_path = join(self.tmp, "/cont_props.h5")
```
Any joined entry starting with `/` will become the new root. I.e. This actually just becomes `/cont_props.h5`. You would want to NOT include the slash
```suggestion
        self.preserve_props_path = join(self.tmp, "cont_props.h5")
```

daltonbohning replied:
```
E.g.
>>> os.path.join("/dir1", "/dir2")
'/dir2'
>>> os.path.join("/dir1", "dir2")
'/dir1/dir2'
```

dsikich replied:
@daltonbohning thanks, good catch. 👍

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

FYI: Errors found in lines not modified in the patch:

[ci/functional/required_packages.sh:6](https://github.com/daos-stack/daos/blob/07e431cadfd4722a01429636eeb2545c90df350c/ci/functional/required_packages.sh#L6):
(lint) client_ver appears unused. Verify use (or export if used externally). [SC2034]

[src/tests/ftest/util/data_mover_test_base.py:1](https://github.com/daos-stack/daos/blob/07e431cadfd4722a01429636eeb2545c90df350c/src/tests/ftest/util/data_mover_test_base.py#L1):
(pylint-too-many-lines) Too many lines in module (1199/1000)

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

FYI: Errors found in lines not modified in the patch:

[ci/functional/required_packages.sh:6](https://github.com/daos-stack/daos/blob/59bb2340a0a8db95c10e3f38f44427014f36d01b/ci/functional/required_packages.sh#L6):
(lint) client_ver appears unused. Verify use (or export if used externally). [SC2034]

[src/tests/ftest/util/data_mover_test_base.py:1](https://github.com/daos-stack/daos/blob/59bb2340a0a8db95c10e3f38f44427014f36d01b/src/tests/ftest/util/data_mover_test_base.py#L1):
(pylint-too-many-lines) Too many lines in module (1199/1000)

ashleypittman commented on utils/rpms/daos.spec:
```diff
@@ -509,13 +509,16 @@ getent passwd daos_agent >/dev/null || useradd -s /sbin/nologin -r -g daos_agent
 # No files in a meta-package
 
 %changelog
+* Mon Nov 03 2021 Danielle M. Sikich <danielle.sikich@intel.com> 2.1.100-6
```
3rd November was a wednesday.

We get lots of these sneak in, and they cause warnings but not failures then every month somebody has to go through and correct them all.

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

FYI: Errors found in lines not modified in the patch:

[ci/functional/required_packages.sh:6](https://github.com/daos-stack/daos/blob/8b4060be30e4799d5056ab30ea23c9a35f2d0149/ci/functional/required_packages.sh#L6):
(lint) client_ver appears unused. Verify use (or export if used externally). [SC2034]

[src/tests/ftest/util/data_mover_test_base.py:1](https://github.com/daos-stack/daos/blob/8b4060be30e4799d5056ab30ea23c9a35f2d0149/src/tests/ftest/util/data_mover_test_base.py#L1):
(pylint-too-many-lines) Too many lines in module (1199/1000)

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

FYI: Errors found in lines not modified in the patch:

[ci/functional/required_packages.sh:6](https://github.com/daos-stack/daos/blob/4f37745775ee6e260d565f7a8054a2d4943bc73e/ci/functional/required_packages.sh#L6):
(lint) client_ver appears unused. Verify use (or export if used externally). [SC2034]

[src/tests/ftest/util/data_mover_test_base.py:1](https://github.com/daos-stack/daos/blob/4f37745775ee6e260d565f7a8054a2d4943bc73e/src/tests/ftest/util/data_mover_test_base.py#L1):
(pylint-too-many-lines) Too many lines in module (1199/1000)

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

FYI: Errors found in lines not modified in the patch:

[ci/functional/required_packages.sh:6](https://github.com/daos-stack/daos/blob/4f37745775ee6e260d565f7a8054a2d4943bc73e/ci/functional/required_packages.sh#L6):
(lint) client_ver appears unused. Verify use (or export if used externally). [SC2034]

[src/tests/ftest/util/data_mover_test_base.py:1](https://github.com/daos-stack/daos/blob/4f37745775ee6e260d565f7a8054a2d4943bc73e/src/tests/ftest/util/data_mover_test_base.py#L1):
(pylint-too-many-lines) Too many lines in module (1199/1000)

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

FYI: Errors found in lines not modified in the patch:

[ci/functional/required_packages.sh:6](https://github.com/daos-stack/daos/blob/4f37745775ee6e260d565f7a8054a2d4943bc73e/ci/functional/required_packages.sh#L6):
(lint) client_ver appears unused. Verify use (or export if used externally). [SC2034]

[src/tests/ftest/util/data_mover_test_base.py:1](https://github.com/daos-stack/daos/blob/4f37745775ee6e260d565f7a8054a2d4943bc73e/src/tests/ftest/util/data_mover_test_base.py#L1):
(pylint-too-many-lines) Too many lines in module (1199/1000)

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

FYI: Errors found in lines not modified in the patch:

[ci/functional/required_packages.sh:6](https://github.com/daos-stack/daos/blob/dab14ee0834f93859e8fef2b659524056ec6eca5/ci/functional/required_packages.sh#L6):
(lint) client_ver appears unused. Verify use (or export if used externally). [SC2034]

[src/tests/ftest/util/data_mover_test_base.py:1](https://github.com/daos-stack/daos/blob/dab14ee0834f93859e8fef2b659524056ec6eca5/src/tests/ftest/util/data_mover_test_base.py#L1):
(pylint-too-many-lines) Too many lines in module (1199/1000)

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

FYI: Errors found in lines not modified in the patch:

[ci/functional/required_packages.sh:6](https://github.com/daos-stack/daos/blob/45922989ef1f7e79b718540412c938190ab1934b/ci/functional/required_packages.sh#L6):
(lint) client_ver appears unused. Verify use (or export if used externally). [SC2034]

[src/tests/ftest/util/data_mover_test_base.py:1](https://github.com/daos-stack/daos/blob/45922989ef1f7e79b718540412c938190ab1934b/src/tests/ftest/util/data_mover_test_base.py#L1):
(pylint-too-many-lines) Too many lines in module (1199/1000)

dsikich commented:
@mchaarawi or @ashleypittman can you re-approve please I keep having to re-push this because I get merge conflicts in the in utils/rpms/daos.spec each time before it gets through CI.

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

FYI: Errors found in lines not modified in the patch:

[ci/functional/required_packages.sh:6](https://github.com/daos-stack/daos/blob/3eb61b6025e7d35ffe0206472aa5d52811b51fdc/ci/functional/required_packages.sh#L6):
(lint) client_ver appears unused. Verify use (or export if used externally). [SC2034]

[src/tests/ftest/util/data_mover_test_base.py:1](https://github.com/daos-stack/daos/blob/3eb61b6025e7d35ffe0206472aa5d52811b51fdc/src/tests/ftest/util/data_mover_test_base.py#L1):
(pylint-too-many-lines) Too many lines in module (1199/1000)

daosbuild1 commented:
Test stage Build RPM on Leap 15 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-6617/54/execution/node/333/log

daosbuild1 commented:
Test stage Build DEB on Ubuntu 20.04 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-6617/54/execution/node/309/log

mchaarawi commented:
> @mchaarawi or @ashleypittman can you re-approve please I keep having to re-push this because I get merge conflicts in the in utils/rpms/daos.spec each time before it gets through CI.

i think master landings are still closed for non 2.0 tickets

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

FYI: Errors found in lines not modified in the patch:

[ci/functional/required_packages.sh:6](https://github.com/daos-stack/daos/blob/3eb61b6025e7d35ffe0206472aa5d52811b51fdc/ci/functional/required_packages.sh#L6):
(lint) client_ver appears unused. Verify use (or export if used externally). [SC2034]

[src/tests/ftest/util/data_mover_test_base.py:1](https://github.com/daos-stack/daos/blob/3eb61b6025e7d35ffe0206472aa5d52811b51fdc/src/tests/ftest/util/data_mover_test_base.py#L1):
(pylint-too-many-lines) Too many lines in module (1199/1000)

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

FYI: Errors found in lines not modified in the patch:

[ci/functional/required_packages.sh:6](https://github.com/daos-stack/daos/blob/2a48831a1c7fc64270d1af57b30ae8ba8c1cd0dc/ci/functional/required_packages.sh#L6):
(lint) client_ver appears unused. Verify use (or export if used externally). [SC2034]

[src/tests/ftest/util/data_mover_test_base.py:1](https://github.com/daos-stack/daos/blob/2a48831a1c7fc64270d1af57b30ae8ba8c1cd0dc/src/tests/ftest/util/data_mover_test_base.py#L1):
(pylint-too-many-lines) Too many lines in module (1301/1000)

[src/tests/ftest/util/data_mover_test_base.py:25](https://github.com/daos-stack/daos/blob/2a48831a1c7fc64270d1af57b30ae8ba8c1cd0dc/src/tests/ftest/util/data_mover_test_base.py#L25):
(pylint-too-many-public-methods) Too many public methods (21/20)

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

FYI: Errors found in lines not modified in the patch:

[ci/functional/required_packages.sh:6](https://github.com/daos-stack/daos/blob/050511edd4132e37486813e92b27d66131dadfff/ci/functional/required_packages.sh#L6):
(lint) client_ver appears unused. Verify use (or export if used externally). [SC2034]

[src/tests/ftest/util/data_mover_test_base.py:1](https://github.com/daos-stack/daos/blob/050511edd4132e37486813e92b27d66131dadfff/src/tests/ftest/util/data_mover_test_base.py#L1):
(pylint-too-many-lines) Too many lines in module (1301/1000)

[src/tests/ftest/util/data_mover_test_base.py:25](https://github.com/daos-stack/daos/blob/050511edd4132e37486813e92b27d66131dadfff/src/tests/ftest/util/data_mover_test_base.py#L25):
(pylint-too-many-public-methods) Too many public methods (21/20)

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

FYI: Errors found in lines not modified in the patch:

[ci/functional/required_packages.sh:6](https://github.com/daos-stack/daos/blob/d4113eba27b26b3d3a3d3bdf17aeda03da66543f/ci/functional/required_packages.sh#L6):
(lint) client_ver appears unused. Verify use (or export if used externally). [SC2034]

[src/tests/ftest/util/data_mover_test_base.py:1](https://github.com/daos-stack/daos/blob/d4113eba27b26b3d3a3d3bdf17aeda03da66543f/src/tests/ftest/util/data_mover_test_base.py#L1):
(pylint-too-many-lines) Too many lines in module (1301/1000)

[src/tests/ftest/util/data_mover_test_base.py:25](https://github.com/daos-stack/daos/blob/d4113eba27b26b3d3a3d3bdf17aeda03da66543f/src/tests/ftest/util/data_mover_test_base.py#L25):
(pylint-too-many-public-methods) Too many public methods (21/20)

daosbuild1 commented:
Test stage Functional Hardware Large completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-6617/64/display/redirect

daosbuild1 commented:
Test stage Functional Hardware Medium completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-6617/64/display/redirect

daosbuild1 commented:
Test stage Functional Hardware Small completed with status FAILURE.  https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-6617/64/display/redirect

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

FYI: Errors found in lines not modified in the patch:

[ci/functional/required_packages.sh:6](https://github.com/daos-stack/daos/blob/9dd1402f017640549d527db1cbd6fb808d78ac40/ci/functional/required_packages.sh#L6):
(lint) client_ver appears unused. Verify use (or export if used externally). [SC2034]

[src/tests/ftest/util/data_mover_test_base.py:1](https://github.com/daos-stack/daos/blob/9dd1402f017640549d527db1cbd6fb808d78ac40/src/tests/ftest/util/data_mover_test_base.py#L1):
(pylint-too-many-lines) Too many lines in module (1301/1000)

[src/tests/ftest/util/data_mover_test_base.py:25](https://github.com/daos-stack/daos/blob/9dd1402f017640549d527db1cbd6fb808d78ac40/src/tests/ftest/util/data_mover_test_base.py#L25):
(pylint-too-many-public-methods) Too many public methods (21/20)

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

FYI: Errors found in lines not modified in the patch:

[ci/functional/required_packages.sh:6](https://github.com/daos-stack/daos/blob/9dd1402f017640549d527db1cbd6fb808d78ac40/ci/functional/required_packages.sh#L6):
(lint) client_ver appears unused. Verify use (or export if used externally). [SC2034]

[src/tests/ftest/util/data_mover_test_base.py:1](https://github.com/daos-stack/daos/blob/9dd1402f017640549d527db1cbd6fb808d78ac40/src/tests/ftest/util/data_mover_test_base.py#L1):
(pylint-too-many-lines) Too many lines in module (1301/1000)

[src/tests/ftest/util/data_mover_test_base.py:25](https://github.com/daos-stack/daos/blob/9dd1402f017640549d527db1cbd6fb808d78ac40/src/tests/ftest/util/data_mover_test_base.py#L25):
(pylint-too-many-public-methods) Too many public methods (21/20)

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

FYI: Errors found in lines not modified in the patch:

[ci/functional/required_packages.sh:6](https://github.com/daos-stack/daos/blob/3cc3e754ebaa0bb6f5d0a9ec1960dc0d9e483f01/ci/functional/required_packages.sh#L6):
(lint) client_ver appears unused. Verify use (or export if used externally). [SC2034]

[src/tests/ftest/util/data_mover_test_base.py:1](https://github.com/daos-stack/daos/blob/3cc3e754ebaa0bb6f5d0a9ec1960dc0d9e483f01/src/tests/ftest/util/data_mover_test_base.py#L1):
(pylint-too-many-lines) Too many lines in module (1301/1000)

[src/tests/ftest/util/data_mover_test_base.py:25](https://github.com/daos-stack/daos/blob/3cc3e754ebaa0bb6f5d0a9ec1960dc0d9e483f01/src/tests/ftest/util/data_mover_test_base.py#L25):
(pylint-too-many-public-methods) Too many public methods (21/20)

ashleypittman commented on src/client/serialize/daos_serialize.c:
```diff
@@ -524,9 +592,79 @@ deserialize_uint(hid_t file_id, uint64_t *val, const char *prop_str)
+		D_ERROR("failed to read property attribute %s\n", prop_str);
+		D_GOTO(out, rc = -DER_MISC);
+	}
+	entry->dpe_val_ptr = (void *)roots;
```
You shouldn't need to cast this.

ashleypittman commented on src/tests/ftest/datamover/posix_preserve_props.py:
```diff
@@ -0,0 +1,240 @@
+        super().setUp()
+
+        # Get the parameters
+        self.ior_flags = self.params.get(
+            "ior_flags", "/run/ior/*")
```
This would probably fit on one line now we have 100 character lines, and is much easier to read.
```suggestion
        self.ior_flags = self.params.get("ior_flags", "/run/ior/*")
```

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

FYI: Errors found in lines not modified in the patch:

[ci/functional/required_packages.sh:6](https://github.com/daos-stack/daos/blob/b1604de820d1c2e90ff2979166024fdbbe98afd7/ci/functional/required_packages.sh#L6):
(lint) client_ver appears unused. Verify use (or export if used externally). [SC2034]

[src/tests/ftest/util/data_mover_test_base.py:1](https://github.com/daos-stack/daos/blob/b1604de820d1c2e90ff2979166024fdbbe98afd7/src/tests/ftest/util/data_mover_test_base.py#L1):
(pylint-too-many-lines) Too many lines in module (1301/1000)

[src/tests/ftest/util/data_mover_test_base.py:25](https://github.com/daos-stack/daos/blob/b1604de820d1c2e90ff2979166024fdbbe98afd7/src/tests/ftest/util/data_mover_test_base.py#L25):
(pylint-too-many-public-methods) Too many public methods (21/20)

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

FYI: Errors found in lines not modified in the patch:

[ci/functional/required_packages.sh:6](https://github.com/daos-stack/daos/blob/7fe3f19fd1a94a637a1b952318fdb5a43433fe6a/ci/functional/required_packages.sh#L6):
(lint) client_ver appears unused. Verify use (or export if used externally). [SC2034]

[src/tests/ftest/util/data_mover_test_base.py:1](https://github.com/daos-stack/daos/blob/7fe3f19fd1a94a637a1b952318fdb5a43433fe6a/src/tests/ftest/util/data_mover_test_base.py#L1):
(pylint-too-many-lines) Too many lines in module (1301/1000)

[src/tests/ftest/util/data_mover_test_base.py:25](https://github.com/daos-stack/daos/blob/7fe3f19fd1a94a637a1b952318fdb5a43433fe6a/src/tests/ftest/util/data_mover_test_base.py#L25):
(pylint-too-many-public-methods) Too many public methods (21/20)

daltonbohning commented on src/tests/ftest/datamover/posix_preserve_props.py:
```diff
@@ -77,7 +77,7 @@ def run_dm_preserve_props(self, tool, cont_type, api):
         result = self.run_datamover(
             self.test_id + " posix to cont2 (empty cont)",
-            "POSIX", posix_path, None, None,
+            "POSIX", posix_path + "/" + self.test_file, None, None,
```
`posix_path` is created above with `posix_path = join(self.new_posix_test_path(), self.test_file)`
So isn't this wrong?

dsikich replied:
@daltonbohning well this was how it was before and is passing because of how ior is run to read/verify the testfile. I think it is confusing yes I can update.

dsikich replied:
@daltonbohning what this is doing is creating a testFile directory on the first copy and it is working because the actual file ior verifies is at <posix_path>/testFile/testFile which is working but weird. I updated it to make it less confusing.

daltonbohning replied:
Ah, I see. Thanks! :)

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

FYI: Errors found in lines not modified in the patch:

[ci/functional/required_packages.sh:6](https://github.com/daos-stack/daos/blob/c98a03c76c3679d75d564a9b4b0d32237aadda95/ci/functional/required_packages.sh#L6):
(lint) client_ver appears unused. Verify use (or export if used externally). [SC2034]

[src/tests/ftest/util/data_mover_test_base.py:1](https://github.com/daos-stack/daos/blob/c98a03c76c3679d75d564a9b4b0d32237aadda95/src/tests/ftest/util/data_mover_test_base.py#L1):
(pylint-too-many-lines) Too many lines in module (1301/1000)

[src/tests/ftest/util/data_mover_test_base.py:25](https://github.com/daos-stack/daos/blob/c98a03c76c3679d75d564a9b4b0d32237aadda95/src/tests/ftest/util/data_mover_test_base.py#L25):
(pylint-too-many-public-methods) Too many public methods (21/20)

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

FYI: Errors found in lines not modified in the patch:

[ci/functional/required_packages.sh:6](https://github.com/daos-stack/daos/blob/22cb02a0f7847d8548e39cc8fb0de3b25cfaa77b/ci/functional/required_packages.sh#L6):
(lint) client_ver appears unused. Verify use (or export if used externally). [SC2034]

[src/tests/ftest/util/data_mover_test_base.py:1](https://github.com/daos-stack/daos/blob/22cb02a0f7847d8548e39cc8fb0de3b25cfaa77b/src/tests/ftest/util/data_mover_test_base.py#L1):
(pylint-too-many-lines) Too many lines in module (1301/1000)

[src/tests/ftest/util/data_mover_test_base.py:25](https://github.com/daos-stack/daos/blob/22cb02a0f7847d8548e39cc8fb0de3b25cfaa77b/src/tests/ftest/util/data_mover_test_base.py#L25):
(pylint-too-many-public-methods) Too many public methods (21/20)
